qa-src/Controllers/User/UserPosts.php

```diff
@@ -27,243 +27,329 @@
+
+		$voteView = qa_get_vote_view('A', false, false);
 
 		foreach ($questions as $question) {
-			$options = qa_post_html_options($question, $htmldefaults);
-			$options['voteview'] = qa_get_vote_view('A', false, false);
+			$options = qa_post_html_options($question, $htmlDefaults);
+			$options['voteview'] = $voteView;
+
+			$qa_content['q_list']['qs'][] = qa_other_to_q_html_fields(
+				$question,
+				$loggedInUserId,
+				qa_cookie_get(),
+				$usersHtml,
+				null,
+				$options
+			);
+		}
+
+		$this->addPageLinks($qa_content, $pagesize, $count);
+		$this->addNavigation($qa_content, $handle, $loggedInUserId, $profileUserId, 'answers');
+
+		return $qa_content;
+	}
+
+	/**
+	 * Return the HTML to display for the handle, and if we're using external users, determine the userid.
+	 *
+	 * @param string $handle
+	 * @param mixed $userId
+	 *
+	 * @return mixed|string
+	 */
+	private function userHtml($handle, $userId)
+	{
+		if (QA_FINAL_EXTERNAL_USERS) {
+			$usersHtml = qa_get_users_html(array($userId), false, qa_path_to_root(), true);
 
-			$qa_content['q_list']['qs'][] = qa_other_to_q_html_fields($question, $loginuserid, qa_cookie_get(),
-				$usershtml, null, $options);
+			return @$usersHtml[$userId];
+		} else {
+			return qa_html($handle);
 		}
+	}
 
-		$qa_content['page_links'] = qa_html_page_links(qa_request(), $start, $pagesize, $count, qa_opt('pages_prev_next'));
+	/**
+	 * @param string $handle
+	 * @param mixed $loggedInUserId
+	 * @param mixed $userId
+	 *
+	 * @return array
+	 */
+	private function fetchActivityDataFromDb($handle, $loggedInUserId, $userId)
+	{
+		$identifier = $this->getGenericUserIdentifier($userId, $handle);
 
+		return qa_db_select_with_pending(
+			qa_db_user_recent_qs_selectspec($loggedInUserId, $identifier, qa_opt_if_loaded('page_size_activity')),
+			qa_db_user_recent_a_qs_selectspec($loggedInUserId, $identifier),
+			qa_db_user_recent_c_qs_selectspec($loggedInUserId, $identifier),
+			qa_db_user_recent_edit_qs_selectspec($loggedInUserId, $identifier)
+		);
+	}
 
-		// Sub menu for navigation in user pages
+	/**
+	 * @param string $handle
+	 * @param mixed $loginuserid
+	 * @param mixed $userId
+	 *
+	 * @return array
+	 */
+	private function fetchQuestionsDataFromDb($handle, $loginuserid, $userId)
+	{
+		$identifier = $this->getGenericUserIdentifier($userId, $handle);
 
-		$ismyuser = isset($loginuserid) && $loginuserid == (QA_FINAL_EXTERNAL_USERS ? $this->userid : $useraccount['userid']);
-		$qa_content['navigation']['sub'] = qa_user_sub_navigation($handle, 'answers', $ismyuser);
+		return qa_db_select_with_pending(
+			qa_db_user_points_selectspec($identifier),
+			qa_db_user_recent_qs_selectspec($loginuserid, $identifier, qa_opt_if_loaded('page_size_qs'), qa_get_start())
+		);
+	}
 
+	/**
+	 * @param string $handle
+	 * @param mixed $loggedInUserId
+	 * @param mixed $userId
+	 *
+	 * @return array
+	 */
+	private function fetchAnswersDataFromDb($handle, $loggedInUserId, $userId)
```

Comment:
You can use type-hints

Reply:
How would you change the function definition to use type-hints?

Reply:
`private function fetchAnswersDataFromDb($handle, int $loggedInUserId, int $userId): array`

Reply:
Would you mind pointing me to a php.net page that states that syntax is valid?

Reply:
http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration - for arguments (PHP 7)

http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration - for return type declaration (PHP 7)

If we works at PHP 5 you cannot use this declarations